sdk/helper/scaleutils/node_identifier.go

```diff
@@ -125,3 +131,59 @@ func filterOutNodeID(n []*api.NodeListStub, id string) []*api.NodeListStub {
 	}
 	return n
 }
+
+// EXPERIMENTAL
+// Node filter options are experimental features and should not be used.
+const (
+	XNodeFilterOptionIgnoreInit  = "node_filter_ignore_init"
+	XNodeFilterOptionIgnoreDrain = "node_filter_ignore_drain"
+)
+
+type NodeFilterOptions struct {
+	ignoreInit  bool
+	ignoreDrain bool
+}
+
+func NewNodeFilterOptions(cfg map[string]string) (*NodeFilterOptions, error) {
+	opts := &NodeFilterOptions{}
+
+	if str, ok := cfg[XNodeFilterOptionIgnoreInit]; ok {
+		ignoreInit, err := strconv.ParseBool(str)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to parse value %s for %s configuration: %v",
+				str, XNodeFilterOptionIgnoreInit, err,
+			)
+		}
+
+		opts.ignoreInit = ignoreInit
+	}
+
+	if str, ok := cfg[XNodeFilterOptionIgnoreDrain]; ok {
+		ignoreDrain, err := strconv.ParseBool(str)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to parse value %s for %s configuration: %v",
+				str, XNodeFilterOptionIgnoreDrain, err,
+			)
+		}
+
+		opts.ignoreDrain = ignoreDrain
+	}
+
+	return opts, nil
+}
+
+func (n *NodeFilterOptions) IgnoreInit() bool {
```

Comment:
`NodeFilterOptions` is a structure only meant to store configurations values, we wont change them in the methods, is it necessary to make it a pointer? We can avoid nil pointers

Reply:
Since I'm still not sure how these options are going to behave the pointer gives us more flexibility to modify things for now. We can just pass and check for `nil` to disable any functionality we don't want to modify at this point.